ApplicationLibCode/Application/Tools/Cloud/RiaSumoConnector.cpp

```diff
@@ -214,6 +255,46 @@ void RiaSumoConnector::requestEnsembleByCasesId( const SumoCaseId& caseId )
              } );
 }
 
+//--------------------------------------------------------------------------------------------------
+///
+//--------------------------------------------------------------------------------------------------
+void RiaSumoConnector::parseEnsembleNames( QNetworkReply* reply, const SumoCaseId& caseId )
+{
+    // See the following file for parsing of the reply
+    // https://github.com/equinor/fmu-sumo/blob/main/src/fmu/sumo/explorer/objects/case.py
+
+    QByteArray result = reply->readAll();
+    reply->deleteLater();
+
+    if ( reply->error() == QNetworkReply::NoError )
+    {
+        m_ensembleNames.clear();
+
+        QJsonDocument doc     = QJsonDocument::fromJson( result );
+        QJsonObject   jsonObj = doc.object();
+        auto          keys_1  = jsonObj.keys();
+
+        auto aggregationsObject = jsonObj["aggregations"].toObject();
+
+        QJsonObject aggregationColumnsObject = aggregationsObject["iteration_names"].toObject();
+        auto        keys_2                   = aggregationColumnsObject.keys();
+
+        QJsonArray bucketsArray = aggregationColumnsObject["buckets"].toArray();
+        foreach ( const QJsonValue& bucket, bucketsArray )
+        {
+            QJsonObject bucketObj = bucket.toObject();
+            auto        keys_3    = bucketObj.keys();
```

Comment:
Mixing `auto` and `QJsonObject` here. I would prefer `QJsonObject` everywhere.

Reply:
Fixed